### PR TITLE
CI: bump pre commit not black

### PR DESCRIPTION
**Pull request by nstarman**
### Description

Partially addresses #15995.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: .ruff.toml [hunk 1]
```diff
@@ -103,9 +103,6 @@ lint.ignore = [
     "PERF203",  # `try`-`except` within a loop incurs performance overhead
     "PERF401",  # Use a list comprehension to create a transformed list
 
-    # pygrep-hooks (PGH)
```
Comment (neutrinoceros) [inline .ruff.toml:106]: these are manual edits, right ? does ruff 0.2 warn about unused ignores ?

**Review by neutrinoceros — APPROVED** [review]
Thank you for doing this !

File: .ruff.toml (hunk 1 above)
Comment (nstarman) [reply .ruff.toml:106]: Yes. Ruff has merged some codes together.

File: astropy/units/tests/test_quantity_non_ufuncs.py [hunk 2]
```diff
@@ -2752,7 +2753,8 @@ def test_known_arguments(self, target, helper):
             name for name in keyword_allowed_helper if not name.startswith("_")
         }
 
-        assert not (diff := keyword_allowed_helper - keyword_allowed_target), (
+        diff = keyword_allowed_helper - keyword_allowed_target
```
Comment (mhvk) [inline astropy/units/tests/test_quantity_non_ufuncs.py:2756]: Is this related? ruff has become anti-walrus? (I'm fine with it, just curious!)

File: astropy/timeseries/periodograms/lombscargle/implementations/fastchi2_impl.py [hunk 3]
```diff
@@ -112,7 +114,9 @@ def lombscargle_fastchi2(
     # Now create an indexing scheme so we can quickly
     # build-up matrices at each frequency
     order = [("C", 0)] if fit_mean else []
-    order.extend(sum(([("S", i), ("C", i)] for i in range(1, nterms + 1)), []))
+    order.extend(
+        chain.from_iterable([("S", i), ("C", i)] for i in range(1, nterms + 1))
+    )
```
[the remark this answers is not in the document]
Comment (mhvk) [reply astropy/timeseries/periodograms/lombscargle/implementations/fastchi2_impl.py:119]: We just have another PR by @eerovaher getting rid of `chain`... :+1: to the above suggestion - nobody has to think what `sum` does to a list or worry about chaining, just explicit what is done.

File: astropy/io/ascii/daophot.py [hunk 4]
```diff
@@ -90,9 +90,7 @@ def parse_col_defs(self, grouped_lines_dict):
         self.col_widths = col_widths
 
         # merge the multi-line header data into single line data
-        coldef_dict = {k: sum(v, []) for (k, v) in coldef_dict.items()}
-
-        return coldef_dict
+        return {k: list(itt.chain.from_iterable(v)) for (k, v) in coldef_dict.items()}
```
Comment (mhvk) [inline astropy/io/ascii/daophot.py:93]: Here and below, I find the `sum` considerably clearer...

**Review by mhvk — COMMENTED** [review]
Mostly looks good, certainly fine for `units` and `utils`.

File: astropy/units/tests/test_quantity_non_ufuncs.py (hunk 2 above)
Comment (neutrinoceros) [reply astropy/units/tests/test_quantity_non_ufuncs.py:2756]: My interpretation is that @nstarman just manually rewrote those lines for readability, and I agree that walruses are hurtful in that regard when they're used to force 2 lines into a single one :)

Comment (eerovaher) [reply astropy/units/tests/test_quantity_non_ufuncs.py:2756]: There is now a rule against assignments in `assert` statements: [RUF018](https://docs.astral.sh/ruff/rules/assignment-in-assert/). It is still allowed to use the walrus elsewhere.

Comment (mhvk) [reply astropy/units/tests/test_quantity_non_ufuncs.py:2756]: Hmm, that rule definitely does not apply here, the variable is very much not used elsewhere!

But anyway, this is fine!

Comment (eerovaher) [reply astropy/units/tests/test_quantity_non_ufuncs.py:2756]: RUF018 forbids `assert ... := ...`, so it definitely does apply here.

File: astropy/coordinates/tests/test_frames.py [hunk 5]
```diff
@@ -1021,8 +1021,10 @@ def test_equal():
     ne = sc1 != sc2
     assert np.all(eq == [True, False])
     assert np.all(ne == [False, True])
-    assert isinstance(v := (sc1[0] == sc2[0]), (bool, np.bool_)) and v
-    assert isinstance(v := (sc1[0] != sc2[0]), (bool, np.bool_)) and not v
+    v = sc1[0] == sc2[0]
+    assert isinstance(v, (bool, np.bool_)) and v
```
[the remark this answers is not in the document]
Comment (dhomeier) [reply astropy/coordinates/tests/test_frames.py:1025]: They already existed in the original version though... but why was that changed in the first place – is there a new rule banning the walrus now?

[the remark this answers is not in the document]
Comment (eerovaher) [reply astropy/coordinates/tests/test_frames.py:1025]: Ruff rule [PT018](https://docs.astral.sh/ruff/rules/pytest-composite-assertion/) forbids using `assert ... and ...` (and `assert not (... or ...)`). We are not enforcing that rule yet, but if we are editing these lines anyways we might as well fix more than one problem.

For the walrus see https://github.com/astropy/astropy/pull/15998#discussion_r1480325736

[the remark this answers is not in the document]
Comment (dhomeier) [reply astropy/coordinates/tests/test_frames.py:1025]: Yes, I fully agreed with that part, and that actually makes it a case where the reasoning behind RUF018 does hold (although I don't recall if we've gotten to really support `-O` in the tests).

File: astropy/units/tests/test_quantity_non_ufuncs.py (hunk 2 above)
Comment (mhvk) [reply astropy/units/tests/test_quantity_non_ufuncs.py:2756]: I meant that the intent of the rule doesn't apply here, `diff` is used only in the exception message.

Comment (nstarman) [reply astropy/units/tests/test_quantity_non_ufuncs.py:2756]: I separated the commit for RUF018 in this PR, so if a majority of folks are against it, it's easily removed. IMO it's largely been an improvement. Keeping `assert` statements very clean, with minimal extra points of failure is a +1. This feels kind of like being careful about what's in the `try` versus `else` parts of a `try-except-else`.

File: astropy/timeseries/periodograms/lombscargle/implementations/fastchi2_impl.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/timeseries/periodograms/lombscargle/implementations/fastchi2_impl.py:119]: If we don't like chain we can use the nested-list comprehension method. This should be faster than for-looping over an `extend`.

``[item for sublist in lists for item in sublist]``

[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/timeseries/periodograms/lombscargle/implementations/fastchi2_impl.py:119]: https://docs.astral.sh/ruff/rules/quadratic-list-summation/

[the remark this answers is not in the document]
Comment (eerovaher) [reply astropy/timeseries/periodograms/lombscargle/implementations/fastchi2_impl.py:119]: Using `chain(()` here is not a bad idea because the number of iterables being combined is variable. We should just avoid the lone `)` if doing so is as simple as it is here.

**Review by dhomeier — CHANGES_REQUESTED** [review]
I am with @mhvk on better getting by without `chain` in `io.ascii`. Applying RUF018 in the cases here and for consistency is fine with me.

File: astropy/units/tests/test_quantity_non_ufuncs.py (hunk 2 above)
Comment (mhvk) [reply astropy/units/tests/test_quantity_non_ufuncs.py:2756]: I think we ended up all agreeing this was an improvement, so no worries!

File: astropy/timeseries/periodograms/lombscargle/implementations/fastchi2_impl.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mhvk) [reply astropy/timeseries/periodograms/lombscargle/implementations/fastchi2_impl.py:119]: Yes, it is possible, but I find @eerovaher's first suggestion much clearer - this is clearly a sin+cos series being created, with possibly a zero-order term.

File: astropy/io/ascii/daophot.py (hunk 4 above)
Comment (mhvk) [reply astropy/io/ascii/daophot.py:93]: Just to have the other options written out
```
return {k: itt.chain(*v) for (k, v) in coldef_dict.items()}
# or
return {k: reduce(operator.iconcat, v, []) for k, v in coldef_dict.items()}
# or, equivalently,
return {k: reduce(list.__iadd__, v, []) for k, v in coldef_dict.items()}
# or really not so bad here (though need better variable names)
return {k: [d for piece in v for d in piece] for k, v in coldef_dict.items()}
```

To be honest, none are as nice as `sum` - they should just optimize it in the standard library! - but oddly I might pick `chain(*v)` - don't know why `from_iterable` just makes it less clear...

Anyway, we're probably overthinking this - large numbers are not going to happen in `sum`

File: astropy/timeseries/periodograms/lombscargle/implementations/fastchi2_impl.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dhomeier) [reply astropy/timeseries/periodograms/lombscargle/implementations/fastchi2_impl.py:119]: And the actual evaluation and summation of orders will definitely take far more time than their set-up.

File: .ruff.toml [hunk 6]
```diff
@@ -103,9 +103,6 @@ lint.ignore = [
...
-    "PGH001",  # eval
```
[the remark this answers is not in the document]
Comment (nstarman) [reply .ruff.toml:107]: ✅

File: astropy/coordinates/tests/test_frames.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (nstarman) [reply astropy/coordinates/tests/test_frames.py:1025]: ✅

File: astropy/io/ascii/daophot.py (hunk 4 above)
Comment (nstarman) [reply astropy/io/ascii/daophot.py]: ✅, using chain.

**Review by mhvk — APPROVED** [review]
Thanks, happy (enough) with this!

File: astropy/io/ascii/daophot.py (hunk 4 above)
Comment (dhomeier) [reply astropy/io/ascii/daophot.py]: That looks best to me, too, if `sum` stays out.

**Review by eerovaher — APPROVED** [review]
I'd prefer the commits to be squashed, but all my suggestions about the final state of the patch have been addressed.

**Comment by nstarman** [comment]
Squash-commit!

**Comment by pllim** [comment]
@dhomeier requested changes, so maybe he should have another look?

File: astropy/io/ascii/daophot.py [hunk 7]
```diff
@@ -181,7 +179,7 @@ def get_cols(self, lines):
                 col.format = fmt
 
         # Set column start and end positions.
-        col_width = sum(self.col_widths, [])
+        col_width = list(itt.chain.from_iterable(self.col_widths))
```
Comment (dhomeier) [inline astropy/io/ascii/daophot.py:182]:
```suggestion
        col_width = list(itt.chain(*self.col_widths))
```
Don't know, did @mhvk's suggestion apply here as well?

**Review by dhomeier — APPROVED** [review]
My requests were about the suggestions from @mhvk already included, just one question remaining for consistency in daophot remaining.
But this is acceptable either way, thanks!

File: astropy/io/ascii/daophot.py (hunk 7 above)
Comment (nstarman) [reply astropy/io/ascii/daophot.py:182]: It can. This was just a 1-liner either way and `chain.from_iterable` is faster than successive packing / unpacking.

Comment (mhvk) [reply astropy/io/ascii/daophot.py:182]: This bothered me less than the other one, maybe because the other had just `v`, so the method really distracted from the variable.

My sense would be that this is fine and not worth running CI for again!

**Comment by nstarman** [comment]
Thanks all!
